api/client/gro_client.py

```diff
@@ -21,62 +29,134 @@
 OUTPUT_FILENAME = 'gro_client_output.csv'
 
 
-def get_df(client, **selected_entities):
-    """Get the content of data series in a pandas frame.
-    selected_entities should be some or all of: item_id, metric_id,
-    region_id, frequency_id, source_id, partner_region_id
-    """
-    return pandas.DataFrame(client.get_data_points(**selected_entities))
+class GroClient(api.client.Client):
+    """A Client with methods to find, and manipulate data series related
+    to a crop and/or region.
 
+    This class offers convenience methods for some common scenarios
+
+    - finding entities by name rather than ids
+    - exploration shortcuts filling in partial selections
+    - finding and saving data series for repeated use, including in a data frame
 
-def print_random_data_series(client, selected_entities):
-    """Example which prints out a CSV of a random data series that
-    satisfies the (optional) given selection.
-    """
-    # Pick a random data series for this selection
-    data_series_list = client.get_data_series(**selected_entities)
-    if not data_series_list:
-        raise Exception("No data series available for {}".format(
-            selected_entities))
-    data_series = data_series_list[int(len(data_series_list)*random())]
-    print("Using data series: {}".format(str(data_series)))
-    print("Outputing to file: {}".format(OUTPUT_FILENAME))
-    writer = unicodecsv.writer(open(OUTPUT_FILENAME, 'wb'))
-    for point in client.get_data_points(**data_series):
-        writer.writerow([point['start_date'], point['end_date'],
-                         point['value'] * point['input_unit_scale'],
-                         client.lookup_unit_abbreviation(point['input_unit_id'])])
-
-
-def search_for_entity(client, entity_type, keywords):
-    """Returns the first result of entity_type (which is items, metrics or
-    regions) that matches the given keywords.
-    """
-    results = client.search(entity_type, keywords)
-    for result in results:
-        print("Picking first result out of {} {}: {}".format(
-            len(results), entity_type, result['id']))
-        return result['id']
-    return None
-
-
-def pick_random_entities(client):
-    """Pick a random item that has some data associated with it, and a
-    random metric and region pair for that item with data
-    available.
     """
-    item_list = client.get_available('items')
-    num = 0
-    while not num:
-        item = item_list[int(len(item_list)*random())]
-        print("Randomly selected item: {}".format(item['name']))
-        selected_entities = {'itemId':  item['id']}
-        entity_list = client.list_available(selected_entities)
-        num = len(entity_list)
-    entities = entity_list[int(num*random())]
-    print("Using entities: {}".format(str(entities)))
-    selected_entities.update(entities)
-    return selected_entities
+
+    _logger = api.client.lib.get_default_logger()
```

Comment:
Are these intentionally static class variables (outside of an __init__)? Could cause confusion if there's ever two instances of GroClient() and one happens to call get_df before the other.

Reply:
done